object_intern.go

```diff
@@ -141,13 +181,16 @@ func (oi *ObjectIntern) AddOrGet(obj []byte, safe bool) (uintptr, error) {
 	// we need to manage it at this layer. Here we add 4 bytes to be used
 	// henceforth as the reference count for this object. Reference count is
 	// always placed as the LAST 4 bytes of an object and is NEVER compressed.
-	objComp = append(objComp, []byte{0x1, 0x0, 0x0, 0x0}...)
-	addr, err = oi.store.Add(objComp)
+	obj = append(obj, InitialRefCount...)
+	addr, err := oi.store.Add(obj)
 	if err != nil {
 		oi.Unlock()
 		return 0, err
 	}
 
+	// remove ref count from end before converting to string
+	objString := string(obj[:len(obj)-4])
```

Comment:
Can probably move this above the append to avoid reslicing obj, not sure that it matters